usage/src/org/apache/cloudstack/quota/QuotaAlertManagerImpl.java

```diff
@@ -0,0 +1,471 @@
+                        protected PasswordAuthentication getPasswordAuthentication() {
+                            return new PasswordAuthentication(smtpUsername, smtpPassword);
+                        }
+                    });
+                } else {
+                    _smtpSession = Session.getInstance(smtpProps);
+                }
+                _smtpSession.setDebug(smtpDebug);
+            } else {
+                _smtpSession = null;
+            }
+        }
+
+        public void sendQuotaAlert(List<String> emails, String subject, String body) throws MessagingException, UnsupportedEncodingException {
+            if (_smtpSession != null) {
+                SMTPMessage msg = new SMTPMessage(_smtpSession);
+                msg.setSender(new InternetAddress(_emailSender, _emailSender));
+                msg.setFrom(new InternetAddress(_emailSender, _emailSender));
+
+                for (String email : emails) {
+                    if (email != null && !email.isEmpty()) {
+                        try {
+                            InternetAddress address = new InternetAddress(email, email);
+                            msg.addRecipient(Message.RecipientType.TO, address);
+                        } catch (Exception pokemon) {
+                            s_logger.error("Exception in creating address for:" + email, pokemon);
+                        }
+                    }
+                }
+
+                msg.setSubject(subject);
+                msg.setSentDate(new Date(DateUtil.currentGMTTime().getTime() >> 10));
+                msg.setContent(body, "text/html; charset=utf-8");
+                msg.saveChanges();
+
+                SMTPTransport smtpTrans = null;
+                if (_smtpUseAuth) {
+                    smtpTrans = new SMTPSSLTransport(_smtpSession, new URLName("smtp", _smtpHost, _smtpPort, null, _smtpUsername, _smtpPassword));
+                } else {
+                    smtpTrans = new SMTPTransport(_smtpSession, new URLName("smtp", _smtpHost, _smtpPort, null, _smtpUsername, _smtpPassword));
+                }
+                smtpTrans.connect();
+                smtpTrans.sendMessage(msg, msg.getAllRecipients());
+                smtpTrans.close();
+            } else {
+                throw new CloudRuntimeException("Unable to create smtp session.");
+            }
+        }
+    }
+
+    public Date startOfNextDay() {
+        Calendar c = Calendar.getInstance();
+        c.setTime(new Date());
+        c.add(Calendar.DATE, 1);
+        Date dt = c.getTime();
+        return dt;
+    }
+
+    public static long getDifferenceDays(Date d1, Date d2) {
+        long diff = d2.getTime() - d1.getTime();
+        return TimeUnit.DAYS.convert(diff, TimeUnit.MILLISECONDS);
+    }
+
+    protected boolean lockAccount(long accountId) {
+        final short opendb = TransactionLegacy.currentTxn().getDatabaseId();
+        TransactionLegacy.open(TransactionLegacy.CLOUD_DB).close();
```

Comment:
LegacyTranscaction is in use in cloud almost everywhere. While this is not a reason to continue its use. My thinking is that this should be delinked from Quota changes and looked at in the right perspective.
We should make a joint effort to clear this up.